content/docs/user-guide/testing/best-practices/_index.md

```diff
@@ -0,0 +1,327 @@
+
+Due to their costs, there should only be a small number of automated system-wide tests in O3DE. System tests should also avoid exhaustively verifying lower-level requirements, as specific issues can be better detected by specific unit and integration tests. The unit and integration tests can filter for known lower-level concerns, with automated system tests as the final least-specific check for unexpected issues.
+
+If you are planning to write a large automated test broadly targeting many systems, please reach out to [SIG-Testing on Discord](https://{{< links/o3de-discord >}}) for design support!
+
+### Acceptance Tests
+
+Acceptance tests focus on the user experience. O3DE has no automated acceptance tests as robots are poorly suited to simulating actual human users. This is due to extreme complexity and cost of Artificial General Intelligence, summed up in observations such as Moravec's Paradox: "it is comparatively easy to make computers exhibit adult level performance on intelligence tests or playing checkers, and difficult or impossible to give them the skills of a one-year-old when it comes to perception and mobility."
+
+O3DE relies on contributors and SIGs to help verify that their code provides a good user experience. This in turn relies on user feedback to inform how the code suits the needs of many different users. Whenever you identify somewhere the code could be improved, please [cut an issue](https://github.com/o3de/o3de/issues/new/choose) or [propose a pull request](/docs/contributing/)!
+
+## Other Automated Test Categories in O3DE
+
+### User Interface (UI) Tests
+
+{{< caution >}}
+Automated UI tests are prone to false failures and false passes. This can lead to the anti-patterns of "when anyone changes any part of the code, they have to change all of the tests" or "this test never catches any real bugs." Due to these risks, only start investing in automated UI tests after automating lower-complexity tests.
+{{< /caution >}}
+
+While there is a significant amount of UI code in O3DE, it's recommended that automated tests avoid targeting the UI layer. UI tests are relatively complex and expensive to maintain. It is also easy to attempt writing acceptance-style tests for the UI, where robotic scripts struggle to provide useful feedback. Such tests can quickly consume more human effort to debug than to manually exercise the UI.
+
+Relying on UI tests to verify features tends to be brittle, and can prompt many confusing debugging sessions. In an ideal scenario there would already be non-UI integration tests, which fully test out the underlying logic of a feature. Then any UI tests above that layer would then not be responsible for testing the system, but only for testing how the interface reacts to simple interactions. Whenever possible, prefer implementing that lower layer of non-UI automation.
+
+### Performance/Load/Stress Tests
```

Comment:
do we consider Soak Testing to be the same as Stress Testing here? Stress being a high transaction/interaction level whereas soak is a long duration test looking for stability and resource leaks.

also do we want to cover Fuzz Testing somewhere. we don't currently do this, but large portions of the engine are involved in deserializing content, settings and other content inputs. Fuzz testing would focus on these parsers or any other interfaces where data enters the engine and can be corrupted.

Reply:
I don't quite know what high-level advice to provide for soak tests, and this isn't trying to present a full taxonomy. I would say that a soak test generally isn't as focused on raw performance metrics, and is often more interested in finding crashes. Perhaps this points toward detecting memory leaks? Is that appropriate for a general testing advice page? Can you recommend a way to expand that into this section?

There's actually a number of lower-level fuzz tests in C++, but I think they're a fairly inefficient way of writing automated tests. They seem like something useful to temporarily execute when you don't actually know what boundaries or failures you need to test against, similar to low-level Chaos Engineering. Again, not quite sure what to recommend about them at a high level, or where to recommend using/avoiding them.

Reply:
perhaps recommend avoiding the inclusion of non-deterministic fuzz testing in general automation. Fuzz testing should be a distinct testing pass using tooling for that purpose. general automation tests should utilize repeatable values rather than changing behaviors so that investigating failures can be accomplished with less effort.

Reply:
perhaps a subsection of "Avoid" titled "Non-Deterministic and Fuzz test values in general tests" then basically say it's tempting to make test values random or use fuzzing for test automation as this seems to cover more ground, however this causes issues investigating any defects encountered as the exact repro may be difficult to generate. reserve fuzz testing for a dedicated pass.